src/modules/RA/DataLayer/components/tabs/MinisheetTab/index.js

```diff
@@ -0,0 +1,119 @@
+  colorPicker: {
+    width: '25%',
+  },
+  addPopup: {
+    display: 'flex',
+    justifyContent: 'center',
+    alignItems: 'center',
+    height: '40vh',
+  },
+  card: {
+    display: 'flex',
+    flexDirection: 'row',
+  },
+  title: {
+    display: 'flex',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+    width: '100%',
+    borderBottom: 'thin solid grey',
+    marginBottom: '25px',
+  },
+});
+
+
+const MinisheetTab = () => {
+  const classes = useStyles();
+  const translate = useTranslate();
+  const { input: { value: fields } } = useField('fields');
+  const {
+    input: {
+      value: {
+        advanced,
+        enable,
+        wizard: { sections = [], title } = {},
+      },
+    },
+  } = useField('minisheet_config');
+
+
+  const getAvailableFields = useCallback(() => {
+    const sectionsFieldIds = sections.reduce((fieldIds, { children = [] }) => {
+      const sectionFieldIds = children.map(({ sourceFieldId }) => sourceFieldId);
+      return [...fieldIds, ...sectionFieldIds];
+    }, []);
+
+    return fields.filter(field => !sectionsFieldIds.find(id => id === field.sourceFieldId));
+  }, [fields, sections]);
+
+  const availableFields = useMemo(() => getAvailableFields(), [getAvailableFields]);
+
+  return (
+    <>
+      {!enable && (
+        <div className={classes.addPopup}>
+          <Card className={classes.card}>
+            <CardContent>
+              <Typography variant="h6">
+                {translate('datalayer.form.popup.card-message')}
+              </Typography>
+            </CardContent>
+            <BooleanInput source="minisheet_config.enable" label="datalayer.form.minisheet.enable" />
+          </Card>
+        </div>
+      )}
+      {enable && (
+        <>
+          <BooleanInput source="minisheet_config.enable" label="datalayer.form.minisheet.enable" />
+          <FieldGroup>
+            <div className={classes.title}>
+              <h3>{translate('datalayer.form.minisheet.title')}</h3>
+              <BooleanInput source="minisheet_config.advanced" label="datalayer.form.minisheet.advanced" />
+            </div>
+            <ColorInput source="highlight_color" label="datalayer.form.minisheet.pick-highlight-color" className={classes.colorPicker} />
+
+            {advanced && (
+            <>
+              <TextInput multiline source="minisheet_template" label="datalayer.form.minisheet.template" fullWidth />
+              <TextInput
+                label="datalayer.form.compare-url.title"
+                source="settings.compare"
+              />
+              <HelpContent title="datalayer.form.compare-url.help-title" content="datalayer.form.compare-url.help-text" />
+            </>
+            )}
+            {!advanced && (
+            <>
+              <MiniSheetFieldTree sections={sections} fields={availableFields} titleField={title} />
+            </>
+            )}
+          </FieldGroup>
+        </>
+      )}
+    </>
+  );
+};
```

Comment:
To avoid large React.Fragment tag you can explicitly return each part like 
```jsx
if(!enable) {
  return  (
    <div className={classes.addPopup}>
       // JSX codes
    </div>
}

if(enable) {
  return  (
    <>
      <BooleanInput source="minisheet_config.enable" label="datalayer.form.minisheet.enable" />
      // JSX codes 
    </>
  )
}
```

But question of taste 🙃

Reply:
Which one is better? I used React.fragment to avoid having multiple return statement but I'm not against your solution if it means more readable code

Reply:
Both are valid, i personally prefer a visual code splitting at first glance

Reply:
Agreed! I split the "rendered" part in two `return` for easier readability